com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -287,20 +289,29 @@ public static void NetworkShow(List<NetworkObject> networkObjects, ulong clientI
                 }
             }
 
-            using (var buffer = PooledNetworkBuffer.Get())
-            using (var writer = PooledNetworkWriter.Get(buffer))
-            {
-                writer.WriteUInt16Packed((ushort)networkObjects.Count);
 
-                for (int i = 0; i < networkObjects.Count; i++)
+            var context = networkManager.MessageQueueContainer.EnterInternalCommandContext(
+                MessageQueueContainer.MessageType.CreateObjects,
+                NetworkChannel.Internal,
+                new[] {clientId},
+                NetworkUpdateLoop.UpdateStage
+            );
+
+            if (context != null)
+            {
+                using (var icontext = (InternalCommandContext) context)
                 {
-                    // Send spawn call
-                    networkObjects[i].Observers.Add(clientId);
+                    icontext.NetworkWriter.WriteUInt16Packed((ushort) networkObjects.Count);
 
-                    networkManager.SpawnManager.WriteSpawnCallForObject(buffer, clientId, networkObjects[i], payload);
-                }
+                    for (int i = 0; i < networkObjects.Count; i++)
```

Comment:
it's a trivial matter of preference, but mine is a `foreach` if there's no need for the indexer variable value - imo it makes the code a tiny bit more readable.

Reply:
I agree with your preference. I'd just point out I didn't write this line, I just changed the indentation on it wrapping it in new stuff, so I'm hesitant to change it since I don't know if there was a reason it was written this way.

Reply:
nit: for is more performant than foreach, so this is fine :P